File: DemoApp Objective-C/CASSample/BannerAdDelegate.m
//
//  BannerAdDelegate.m
//  CASSample
//
//  Created by Денис Ешенко on 20.06.2022.
//

#import "BannerAdDelegate.h"

@implementation BannerAdDelegate
- (void)bannerAdViewDidLoad:(CASBannerView *)view {
	if(self.infoLabel) {
		self.infoLabel.text = @"Ready";
	}
    NSLog(@"[CAS Sample] Banner Ad loaded and ready to present");
}
- (void)bannerAdView:(CASBannerView *)adView didFailWith:(enum CASError)error {
	NSString * message = [NSString stringWithFormat: @"Error code %ld", (long)error];
	if(self.infoLabel) {
		self.infoLabel.text = message;
	}
	NSLog(@"[CAS Sample] Banner Ad did fail: %@", message);
}
- (void)bannerAdView:(CASBannerView *)adView willPresent:(id<CASStatusHandler>)impression {
    NSLog(@"[CAS Sample] Banner Ad will present");
}
- (void)bannerAdViewDidRecordClick:(CASBannerView *)adView {
    NSLog(@"[CAS Sample] Banner Ad did record click");
}
@end

File: Swift/CASSample/BannerAdDelegate.swift
//
//  BannerAdDelegate.swift
//  CASSample
//
//  Copyright © 2022 Clever Ads Solutions. All rights reserved.
//

import CleverAdsSolutions
import Foundation
import UIKit

class BannerAdDelegate: CASBannerDelegate {
    var infoLabel: UILabel?

    func bannerAdViewDidLoad(_ view: CASBannerView) {
        infoLabel?.text = "Ready"
        print("[CAS Sample] Banner Ad loaded and ready to present")
    }

    func bannerAdView(_ adView: CASBannerView, didFailWith error: CASError) {
        infoLabel?.text = error.message
        print("[CAS Sample] Banner Ad did fail: \(error.message)")
    }

    func bannerAdView(_ adView: CASBannerView, willPresent impression: CASImpression) {
        print("[CAS Sample] Banner Ad will present")
    }

    func bannerAdViewDidRecordClick(_ adView: CASBannerView) {
        print("[CAS Sample] Banner Ad did record click")
    }
}
